### PR TITLE
CDK: Fix class SessionTokenAuthenticator for CLASS_TYPES_REGISTRY mapper

### PR DESCRIPTION
**Pull request by yevhenii-ldv**
## What
*Describe what the change is solving*
*It helps to add screenshots if it affects the frontend.*

## How
*Change `SessionTokenAuthenticator` to `LegacySessionTokenAuthenticator`*

## Recommended reading order
1. `airbyte-cdk/python/airbyte_cdk/sources/declarative/parsers/class_types_registry.py`

### Airbyter

Before merging:
- Pull Request description explains what problem it is solving
- Code change is unit tested
- Build and my-py check pass
- Smoke test the change on at least one affected connector
   - On Github: Run [this workflow](https://github.com/airbytehq/airbyte/actions/workflows/connectors_tests.yml), passing `--use-local-cdk --name=source-<connector>` as options
   - Locally: `airbyte-ci connectors --use-local-cdk --name=source-<connector> test`
- PR is reviewed and approved
      
After merging:
- [Publish the CDK](https://github.com/airbytehq/airbyte/actions/workflows/publish-cdk-command-manually.yml)
   - The CDK does not follow proper semantic versioning. Choose minor if this the change has significant user impact or is a breaking change. Choose patch otherwise.
   - Write a thoughtful changelog message so we know what was updated.
- Merge the platform PR that was auto-created for updating the Connector Builder's CDK version
   - This step is optional if the change does not affect the connector builder or declarative connectors.

</details>

### CODE REVIEW

File: airbyte-cdk/python/airbyte_cdk/sources/declarative/parsers/class_types_registry.py [hunk 1]
```diff
@@ -56,7 +56,7 @@
 """
 CLASS_TYPES_REGISTRY contains a mapping of developer-friendly string -> class to abstract the specific class referred to
 """
-CLASS_TYPES_REGISTRY: Mapping[str, Type] = {
+CLASS_TYPES_REGISTRY: Mapping[str, type] = {
```
Comment (girarda) [inline airbyte-cdk/python/airbyte_cdk/sources/declarative/parsers/class_types_registry.py:59]: why this change?

Comment (yevhenii-ldv) [reply airbyte-cdk/python/airbyte_cdk/sources/declarative/parsers/class_types_registry.py:59]: I've answered it here: https://github.com/airbytehq/airbyte/pull/31696#discussion_r1378748100

**Review by girarda — APPROVED** [review]
:shipit: thanks for fixing!
